chain/btcd.go

```diff
@@ -197,6 +197,44 @@ func (c *RPCClient) BlockStamp() (*waddrmgr.BlockStamp, error) {
 	}
 }
 
+// maybeShouldFetchBlock returns true if the contents of a block *might* have
+// some items that match our watch list. This uses the neutrino filters to do a
+// quick loop up to see if the block has relevant items to avoid downloading
+// the full block.
+func maybeShouldFetchBlock(rpc *rpcclient.Client, blk wtxmgr.BlockMeta,
+	watchList [][]byte) (bool, error) {
+
+	rawFilter, err := rpc.GetCFilter(&blk.Hash, wire.GCSFilterRegular)
```

Comment:
If block filters are turned off in `bitcoind`, will this return an error? Maybe we need to do a feature discovery first.

Reply:
True, we only want to try if the bitcoind node has it.

Reply:
Updated! Turns out bitcoind also has a different name for the RPC to fetch the filter, so added another layer of abstraction to handle that.